Sockets cleanup

hovi opened the pull request:
- SpaceEngineers tests are closing connection correctly after each run.
- SocketEnvironment closing connection is safer.
- Timeouts for connecting and idle connection in SocketEnvironment.

Could be refactored better (especially tests), but if we move to kotlin, I don't think it's worth the effort.

PremekPaska committed:
Merge branch 'SeGrid' into se-dev

PremekPaska commented on src/main/java/spaceEngineers/environments/SocketEnvironment.java:
```diff
@@ -32,6 +33,8 @@
 
         int maxWaitTime = 20000;
+        int socketConnectionTimeout = 4000;
+        int socketDataTimeout = 4000;
```
It's not critical, but I like when physical quantities have units in the name of the variable, such as:

maxWaitTimeMs
socketConnectionTimeoutMs
etc.

PremekPaska replied:
Just a Java newbie question: why not add something like final / const?

hovi replied:
I renamed variables to match this.

Yes good point, final should be in 99% places of the java code, including method parameters. There's even settings in IDEA (or Eclipse), that adds it all automatically on save. I used to do that before switching to kotlin.
These variables don't make sense at all and they should be (final) constructor parameters with some sane defaults. imho :-)
In kotlin, 99% variables should be "val", which serves the same purpose.

PremekPaska reviewed (approved):
Awesome :)

I added one suggestion and one question.

PremekPaska reviewed (approved):
OK, so please combine all 3 commits that are related to opening connection / timeouts into one commit and merge it.

hovi committed:
Bugfixes related to TCP networking.

Making sure there are timeouts for socket (connection/idle).
Closing environment connection after each test.
Specifically set stream encoding to UTF8. Otherwise JVM uses platform-specific.

PremekPaska commented:
Merged manually (fast forward, since it's only one commit).
